Firestore/core/src/local/leveldb_transaction.cc

```diff
@@ -200,6 +201,33 @@ void LevelDbTransaction::Delete(absl::string_view key) {
   version_++;
 }
 
+bool LevelDbTransaction::DeleteEverythingWithPrefix(
```

Comment:
This function is inspired from `DeleteEverythingWithPrefix` from `Firestore/core/src/local/leveldb_migrations.cc`.

The reference function does Database migration and set the maximum capacity of transaction to 1000 operation. 

For other functions in indexManager, the `commit transaction` is handled in `persistence_->Run`, so it is one commit per function.

It is not clear why this `1000` restriction is place as transaction size, not operation size inside  `commit`. Therefore the safest way to implement `deletedAllIndexes` is to use multiple transaction to execute the task.

Reply:
* Can you extract the implementation in leveldb_migration to somewhere, and reuse it here? Also, let's make 1000 a constant, instead of a magic number.
* Optional: Both your implementation and the original one decode every key, which could be a lot. This can be done more efficiently using `PrefixSuccessor` in `string_util.cc`. You would hold a start cursor, and an end cursor, delete all the way until start meets end.